### DISCUSSION TITLE
How to include where condition for relation field in custom function?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2086] [Q&A]
Hi I just need to sum two fields of a ledger in custom function which is not properly linking with struct

```
var res struct {
Cr float64 `json:"cr"`
Dr float64 `json:"dr"`
}

// Method 1
q := txDao.DB().NewQuery("SELECT sum(cr) as Cr, sum(dr) as Dr FROM {{ledgers}} WHERE [[bank]]={:id}")
q.Bind(dbx.Params{"id": id})
if err := q.One(&res); err != nil {
return err
}
// Method 2
txDao.DB().Select("sum(cr) as Cr", "sum(dr) as Dr").From("ledgers").Where(dbx.HashExp{"[[bank.id]]": id}).Row(&res)
```

I have tried column name as [[bank]] as well as [[bank.id]] both not working... can you please let me know any other mistake i have done?

**Comment by ganigeorgiev** [comment]
The question lacks context and I'm not sure that I understand the example.

Does the query return any error?

What is `bank`? Is it another collection or field in `ledgers`? If it is a field - is it `single` or `multiple` relation (aka. "Max Select != 1")? If it is multiple, then the stored value is a serialized json array, and you have to use [`json_each`](https://www.sqlite.org/json1.html#jeach) if you want to apply a condition on each value.

Additionally, if you have custom column names, you can try specifying an explicit `db:"yourColumn"` struct tag for each field.

**Reply by iamdineshkumar** [reply]
there are two collections banks & ledger. In ledger there is a field bank which is of type relation to banks table which is single relation. Based on your feedback i have changed struct tag and tried with running query below mentioned way, Its not giving any error as well. 
```
var res struct {
	Cr float64 `json:"cr" db:"Cr"`
	Dr float64 `json:"dr" db:"Dr"`
}
//Method -1 
txDao.DB().Select("sum(cr) as Cr", "sum(dr) as Dr").From("ledgers").Where(dbx.HashExp{"[[bank]]": id}).Row(&res)
//Method -2 
txDao.DB().Select("sum(cr) as Cr", "sum(dr) as Dr").From("ledgers").Where(dbx.HashExp{"[[bank.id]]": id}).Row(&res)
```

Note: Actually I started using go lang for past 24 hours only :P ... So my learning of go lang in a day can have mistakes, so if any mistake on my code also let me know...

**Reply by ganigeorgiev** [reply]
Hm, it is possible that the same sum column names are interfering with the struct scanning, I'm not sure. You can try:

```go
var res struct {
	Cr float64 `json:"cr" db:"crsum"`
	Dr float64 `json:"dr" db:"drsum"`
}

err := app.Dao().DB().
	Select("sum(cr) as crsum", "sum(dr) as drsum").
	From("ledgers").
	Where(dbx.HashExp{"bank": "YOUR_BANK_ID"}).
	One(&res)

log.Println("Error:", err)
```

Or with `Row()`:

```go
var cr, dr float64

err := app.Dao().DB().
	Select("sum(cr)", "sum(dr)").
	From("ledgers").
	Where(dbx.HashExp{"bank": "YOUR_BANK_ID"}).
	Row(&cr, &dr)

log.Println("Error:", err)
```

If that doesn't help, please provide a minimal reproducible repo or at least a more detailed code sample.

**Reply by iamdineshkumar** [reply]
Thanks it works....
